conda_anaconda_telemetry/hooks.py

```diff
@@ -238,11 +241,22 @@ def _conda_request_headers() -> Sequence[HeaderWrapper]:
     return custom_headers
 
 
+@functools.lru_cache(None)
```

Comment:
I don't think you should use `functools.lru_cache` here. I seriously doubt you are going to see the same `(host, path)` pair too many times. The real caching is in the header construction itself